README.md

```diff
@@ -22,9 +24,66 @@ wayland exclusive services
 * `mako.service`
 * `swayidle.service`
 
+## How this works
+
+Here's a summary of the flow of how this works. See the linked files for details.
+
+A [`sway.desktop`](./wayland-sessions/sway-session.desktop) file will be installed in a `wayland-sessions` directory. Desktop managers
+like `gdm` will detect this and use it to present Sway Service as a menu option.
+
+Once selected, the provided [`sway-user-service`](./bin/sway-user-service) script will be executed.
+This script:
+   1. sources `/etc/profile`
+   2. sets `XDG_CURRENT_DESKTOP=sway`
+   3. exports environment variables to the user's systemd instance
+   4. Starts the provided `sway.service`
+
+[`sway.service`](./systemd/sway.service) does the following:
+
+  1. Runs all the units associated with `sway-session-pre.target`
+  2. Sources ~/.config/sway/env
+  3. Runs `sway` binary
+  4. Runs all units associated with `sway-session.target` (See "Provided services" above.)
+
+When the `sway` binary runs, it will run the [`config/sway/10-service.conf`] file that you must
+load yourself. This handles loading environment variables that Sway sets back into `systemd`'s user
+environment.
+
+The provided services are not enabled by default. If you want enable `mako` example,
```

Comment:
I apologize since I wasn't active the previous years. could you replace `mako` by a provided unit please ?